File: backend/src/main/java/com/events/model/Event.java
package com.events.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Entity
@Builder
@AllArgsConstructor
@Table(name = "events")
public class Event {
    @Id
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column
    private String description;

    @Column(nullable = false)
    private String category;

    @Column(nullable = false)
    private LocalDateTime date;

    @Column(nullable = false)
    private String venue;

    private String address;

    private String coverImage;

    private Double price;

    private String currency;

    private Integer totalSpots;

    private Integer spotsRemaining;

    @Embedded
    @AttributeOverrides({
            @AttributeOverride(name = "name", column = @Column(name = "organizer_name")),
            @AttributeOverride(name = "email", column = @Column(name = "organizer_email"))
    })
    private  Organizer organizer;

    public Event() {
        this.id = UUID.randomUUID();
    }


}

File: frontend/lib/api/event.ts
import { Base, PaginatedBase } from "./base";
import { get } from "./client";

export interface Event {
  id: string;
  name: string;
  description: string;
  category: "Music" | "Tech" | "Food" | "Art" | "Fitness";
  date: string;
  venue: string;
  address: string;
  coverImage: string;
  price: number;
  currency: string;
  totalSpots: number;
  spotsRemaining: number;
  organizerName: string;
  organizerEmail: string;
  priceDisplay: string;
}

export type EventSummary = Pick<
  Event,
  | "id"
  | "name"
  | "date"
  | "venue"
  | "category"
  | "coverImage"
  | "priceDisplay"
  | "price"
>;

export const eventApi = {
  getAll: (
    page = 0,
    limit = 9,
    search = "",
    category = "",
    dateFrom?: string,
    dateTo?: string,
  ) =>
    get<PaginatedBase<EventSummary[]>>(
      `/events?page=${page}&limit=${limit}${search ? `&search=${search}` : ""}${category ? `&category=${category}` : ""}${dateTo ? `&dateTo=${dateTo}` : ""}${dateFrom ? `&dateFrom=${dateFrom}` : ""}`,
    ),
  getSingle: (id: string) => get<Base<Event>>(`/events/${id}`),
};
